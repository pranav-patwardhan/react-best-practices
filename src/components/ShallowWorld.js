import React, { Component } from 'react';
class ShallowWorld extends Component{

    render(){
        return(
            <div className="container">
                <strong>This is just a test for shallow {this.props.name}</strong>
            </div>
        );
    }
}

export default ShallowWorld;