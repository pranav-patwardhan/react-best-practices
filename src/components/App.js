import React, {Component} from 'react';
import logo from '../assets/images/logo.svg';
import ShallowWorld from './ShallowWorld';
import '../assets/css/App.css';
let data = [];

class App extends Component {
    constructor(props) {
        super(props);
        this.state = {
            homeList: []
        };
    }

    componentWillMount() {
        // this.getAll();
    }

    getAll() {

        let currentThis = this;
        this.props.getAllAssist()
            .then(function (res) {
                data = res.payload.data;
                currentThis.setState({homeList: data});
            }, function (err) {
                console.log("res", err);
                alert(err)
            })

    }
    changePage(){
        alert("herllo");
        this.props.changePage();
    }
    render() {
        console.log(this.state.homeList)
        return (
            <div className="App">
                <div>
                    <h1>Home</h1>
                    <p>Welcome home!</p>
                    <button onClick={() => this.props.changePage()}>Go to about page via redux</button>
                </div>

                <ShallowWorld name="Surj"/>
                <ShallowWorld name="Al"/>
                <button className="checkClick" onClick={() => this.getAll()}>Get Assist Data</button>
                <ul>
                    <li>{this.state.homeList.id}</li>
                    <li>{this.state.homeList.title}</li>
                    <li>{this.state.homeList.body}</li>
                </ul>
            </div>
        );
    }
}
export default App;
