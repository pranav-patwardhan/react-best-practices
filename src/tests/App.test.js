import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import App from '../components/App';
import ShallowWorld from '../components/ShallowWorld'



describe('<App />', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<App />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders and matches our snapshot', () => {
        const component = renderer.create(
            <App  />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('should render 2 Shalloworld components', () => {
        const wrapper = shallow(<App name="app"/>);

            expect(wrapper.find(ShallowWorld)).toHaveLength(2);
    });
    describe('should render 1 components', () => {
        const component = shallow(<App name="app"/>);
        console.log(component.instance().props);
        expect(component.instance().props.name).toBe('app');
    });


});