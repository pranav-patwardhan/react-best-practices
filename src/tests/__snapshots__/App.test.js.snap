// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders and matches our snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to Loomi UI
    </h1>
  </header>
  <p
    className="App-intro"
  >
    Demo implementation 
    <code>
      Redux
    </code>
     Happy Coding :).
  </p>
  <div>
    <strong>
      This is just a test for shallow 
      Surj
    </strong>
  </div>
  <div>
    <strong>
      This is just a test for shallow 
      Al
    </strong>
  </div>
  <button
    className="checkClick"
    onClick={[Function]}
  >
    Get Assist Data
  </button>
  <ul>
    <li />
    <li />
    <li />
  </ul>
</div>
`;

exports[`[object Object] renders and matches our snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to Loomi UI
    </h1>
  </header>
  <p
    className="App-intro"
  >
    Demo implementation 
    <code>
      Redux
    </code>
     Happy Coding :).
  </p>
  <div>
    <strong>
      This is just a test for shallow 
      Pranav
    </strong>
  </div>
  <button
    className="checkClick"
    onClick={[Function]}
  >
    Get Assist Data
  </button>
  <ul>
    <li />
    <li />
    <li />
  </ul>
</div>
`;

exports[`class App extends _react.Component {
    constructor(props) {
        super(props);
        this.state = {
            virtualList: [] };

    }

    componentWillMount() {
        // this.getAll();
    }

    getAll() {

        let currentThis = this;
        this.props.getAllAssist().
        then(function (res) {
            data = res.payload.data;
            currentThis.setState({ virtualList: data });
        }, function (err) {
            console.log("res", err);
            alert(err);
        });

    }

    render() {
        console.log(this.state.virtualList);
        return (
            _react2.default.createElement('div', { className: 'App', __source: { fileName: _jsxFileName, lineNumber: 35 }, __self: this },
                _react2.default.createElement('header', { className: 'App-header', __source: { fileName: _jsxFileName, lineNumber: 36 }, __self: this },
                    _react2.default.createElement('img', { src: _logo2.default, className: 'App-logo', alt: 'logo', __source: { fileName: _jsxFileName, lineNumber: 37 }, __self: this }),
                    _react2.default.createElement('h1', { className: 'App-title', __source: { fileName: _jsxFileName, lineNumber: 38 }, __self: this }, 'Welcome to Loomi UI')),

                _react2.default.createElement('p', { className: 'App-intro', __source: { fileName: _jsxFileName, lineNumber: 40 }, __self: this }, 'Demo implementation ',
                    _react2.default.createElement('code', { __source: { fileName: _jsxFileName, lineNumber: 41 }, __self: this }, 'Redux'), ' Happy Coding :).'),

                _react2.default.createElement('button', { className: 'checkClick', onClick: () => this.getAll(), __source: { fileName: _jsxFileName, lineNumber: 43 }, __self: this }, 'Get Assist Data'),
                _react2.default.createElement('ul', { __source: { fileName: _jsxFileName, lineNumber: 44 }, __self: this },
                    _react2.default.createElement('li', { __source: { fileName: _jsxFileName, lineNumber: 45 }, __self: this }, this.state.virtualList.id),
                    _react2.default.createElement('li', { __source: { fileName: _jsxFileName, lineNumber: 46 }, __self: this }, this.state.virtualList.title),
                    _react2.default.createElement('li', { __source: { fileName: _jsxFileName, lineNumber: 47 }, __self: this }, this.state.virtualList.body))));



    }} renders and matches our snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to Loomi UI
    </h1>
  </header>
  <p
    className="App-intro"
  >
    Demo implementation 
    <code>
      Redux
    </code>
     Happy Coding :).
  </p>
  <button
    className="checkClick"
    onClick={[Function]}
  >
    Get Assist Data
  </button>
  <ul>
    <li />
    <li />
    <li />
  </ul>
</div>
`;

exports[`class App extends _react.Component {
    constructor(props) {
        super(props);
        this.state = {
            virtualList: [] };

    }

    componentWillMount() {
        // this.getAll();
    }

    getAll() {

        let currentThis = this;
        this.props.getAllAssist().
        then(function (res) {
            data = res.payload.data;
            currentThis.setState({ virtualList: data });
        }, function (err) {
            console.log("res", err);
            alert(err);
        });

    }

    render() {
        console.log(this.state.virtualList);
        return (
            _react2.default.createElement('div', { className: 'App', __source: { fileName: _jsxFileName, lineNumber: 36 }, __self: this },
                _react2.default.createElement('header', { className: 'App-header', __source: { fileName: _jsxFileName, lineNumber: 37 }, __self: this },
                    _react2.default.createElement('img', { src: _logo2.default, className: 'App-logo', alt: 'logo', __source: { fileName: _jsxFileName, lineNumber: 38 }, __self: this }),
                    _react2.default.createElement('h1', { className: 'App-title', __source: { fileName: _jsxFileName, lineNumber: 39 }, __self: this }, 'Welcome to Loomi UI')),

                _react2.default.createElement('p', { className: 'App-intro', __source: { fileName: _jsxFileName, lineNumber: 41 }, __self: this }, 'Demo implementation ',
                    _react2.default.createElement('code', { __source: { fileName: _jsxFileName, lineNumber: 42 }, __self: this }, 'Redux'), ' Happy Coding :).',
                    _react2.default.createElement(_ShallowWorld2.default, { name: 'Pranav', __source: { fileName: _jsxFileName, lineNumber: 43 }, __self: this })),

                _react2.default.createElement('button', { className: 'checkClick', onClick: () => this.getAll(), __source: { fileName: _jsxFileName, lineNumber: 45 }, __self: this }, 'Get Assist Data'),
                _react2.default.createElement('ul', { __source: { fileName: _jsxFileName, lineNumber: 46 }, __self: this },
                    _react2.default.createElement('li', { __source: { fileName: _jsxFileName, lineNumber: 47 }, __self: this }, this.state.virtualList.id),
                    _react2.default.createElement('li', { __source: { fileName: _jsxFileName, lineNumber: 48 }, __self: this }, this.state.virtualList.title),
                    _react2.default.createElement('li', { __source: { fileName: _jsxFileName, lineNumber: 49 }, __self: this }, this.state.virtualList.body))));



    }} renders and matches our snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to Loomi UI
    </h1>
  </header>
  <p
    className="App-intro"
  >
    Demo implementation 
    <code>
      Redux
    </code>
     Happy Coding :).
    <div>
      This is just a test for shallow 
      Pranav
    </div>
  </p>
  <button
    className="checkClick"
    onClick={[Function]}
  >
    Get Assist Data
  </button>
  <ul>
    <li />
    <li />
    <li />
  </ul>
</div>
`;

exports[`class App extends _react.Component {
    constructor(props) {
        super(props);
        this.state = {
            virtualList: [] };

    }

    componentWillMount() {
        // this.getAll();
    }

    getAll() {

        let currentThis = this;
        this.props.getAllAssist().
        then(function (res) {
            data = res.payload.data;
            currentThis.setState({ virtualList: data });
        }, function (err) {
            console.log("res", err);
            alert(err);
        });

    }

    render() {
        console.log(this.state.virtualList);
        return (
            _react2.default.createElement('div', { className: 'App', __source: { fileName: _jsxFileName, lineNumber: 37 }, __self: this },
                _react2.default.createElement('header', { className: 'App-header', __source: { fileName: _jsxFileName, lineNumber: 38 }, __self: this },
                    _react2.default.createElement('img', { src: _logo2.default, className: 'App-logo', alt: 'logo', __source: { fileName: _jsxFileName, lineNumber: 39 }, __self: this }),
                    _react2.default.createElement('h1', { className: 'App-title', __source: { fileName: _jsxFileName, lineNumber: 40 }, __self: this }, 'Welcome to Loomi UI')),

                _react2.default.createElement('p', { className: 'App-intro', __source: { fileName: _jsxFileName, lineNumber: 42 }, __self: this }, 'Demo implementation ',
                    _react2.default.createElement('code', { __source: { fileName: _jsxFileName, lineNumber: 43 }, __self: this }, 'Redux'), ' Happy Coding :).'),

                _react2.default.createElement(_ShallowWorld2.default, { name: 'Pranav', __source: { fileName: _jsxFileName, lineNumber: 45 }, __self: this }),
                _react2.default.createElement('button', { className: 'checkClick', onClick: () => this.getAll(), __source: { fileName: _jsxFileName, lineNumber: 46 }, __self: this }, 'Get Assist Data'),
                _react2.default.createElement('ul', { __source: { fileName: _jsxFileName, lineNumber: 47 }, __self: this },
                    _react2.default.createElement('li', { __source: { fileName: _jsxFileName, lineNumber: 48 }, __self: this }, this.state.virtualList.id),
                    _react2.default.createElement('li', { __source: { fileName: _jsxFileName, lineNumber: 49 }, __self: this }, this.state.virtualList.title),
                    _react2.default.createElement('li', { __source: { fileName: _jsxFileName, lineNumber: 50 }, __self: this }, this.state.virtualList.body))));



    }} renders and matches our snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to Loomi UI
    </h1>
  </header>
  <p
    className="App-intro"
  >
    Demo implementation 
    <code>
      Redux
    </code>
     Happy Coding :).
  </p>
  <div>
    <strong>
      This is just a test for shallow 
      Pranav
    </strong>
  </div>
  <button
    className="checkClick"
    onClick={[Function]}
  >
    Get Assist Data
  </button>
  <ul>
    <li />
    <li />
    <li />
  </ul>
</div>
`;
